File: ScaleformUI_Csharp/Elements/ScaleformLabel.cs
namespace ScaleformUI.Elements
{
    public struct ScaleformLabel
    {
        private readonly string _label;

        public ScaleformLabel(string label)
        {
            _label = label;
        }

        public string Label
        {
            get
            {
                return _label;
            }
        }

        public string[] SplitLabel
        {
            get
            {
                int stringsNeeded = (_label.Length - 1) / 99 + 1; // division with round up

                String[] outputString = new String[stringsNeeded];
                for (int i = 0; i < stringsNeeded; i++)
                {
                    outputString[i] = _label.Substring(i * 99, MathUtil.Clamp(_label.Substring(i * 99).Length, 0, 99));
                }

                return outputString;
            }
        }

        public static implicit operator ScaleformLabel(string label)
        {
            return new ScaleformLabel(label);
        }

        public static bool operator ==(ScaleformLabel a, ScaleformLabel b)
        {
            return a.Label == b.Label;
        }

        public static bool operator !=(ScaleformLabel a, ScaleformLabel b)
        {
            return a.Label != b.Label;
        }

        public override bool Equals(object obj)
        {
            return obj is ScaleformLabel label && this == label;
        }

        public override int GetHashCode()
        {
            return Label.GetHashCode();
        }
    }
}

File: ScaleformUI_V8/src/elements/scaleform-label.ts
export class ScaleformLabel {
    private readonly _label: string;

    constructor(label: string) {
        this._label = label;
    }

    get Label(): string {
        return this._label;
    }

    equals(obj: Object): boolean {
        if (!(obj instanceof ScaleformLabel)) {
            return false;
        }
        const otherLabel = obj as ScaleformLabel;
        return this.Label === otherLabel.Label;
    }
}

function implicitOperator(label: string): ScaleformLabel {
    return new ScaleformLabel(label);
}

function operatorEquals(a: ScaleformLabel, b: ScaleformLabel): boolean {
    return a.equals(b);
}

function operatorNotEquals(a: ScaleformLabel, b: ScaleformLabel): boolean {
    return !operatorEquals(a, b);
}